databroker/assets/core.py

```diff
@@ -37,7 +37,7 @@ def _get_datum_from_datum_id(col, datum_id, datum_cache, logger):
     try:
         datum = datum_cache[datum_id]
     except KeyError:
-        keys = ['datum_kwargs', 'resource']
+        keys = ['datum_kwargs', 'resource', 'datum_id']
```

Comment:
I assume this is so that you can get the full datum out later?

Reply:
No this was so we have parity between what the `RunEngine` puts out and what the `Broker` puts out. On line https://github.com/NSLS-II/databroker/pull/379/files#diff-153ae1bc82a20d75fad0da96762b32d5R47 the only things in the datum are in those keys.

Reply:
That is what I meant.  Most of this code was written assuming that datum/resource were not going to make it out to the world.  The reason the `datum_id` was not in this originally is that if you were asking for the resource and datum_kwargs, you already had the datum_id so why store it again.

Might be worth looking at just using `edoc` directly (and popping off `_id`) rather than using `keys` to construct a new dictionary.